### ISSUE TITLE
Error while defining a map of object

### ISSUE THREAD

**Issue by lahirus** [ballerina-platform/ballerina-lang#9883]
**Description:**
I get an error with this program. Error is not clear and no ballerina-internal.log created as well.

Error: 
error: error, message: 5
    at student_info:0.0.0:student_info:0.0.0.<init>(student_info_service.bal:6)

Code:

```ballerina
import ballerina/io;
import ballerina/http;
import ballerina/log;

map<Student> studentMap = {
    "0989898": new Student(0989898, "34534253", "Edwin", "Tyler", "09/01/2006", 2012, true),
    "45675647": new Student(45675647, "34534253", "Angele", "Marburger", "09/02/2006", 2012, true),
    "78687765": new Student(78687765, "34534253", "Ming", "Gower", "09/03/2006", 2013, true),
    "54667677": new Student(54667677, "34534253", "Ervin", "Rappa", "09/04/2006", 2013, false),
    "00676676": new Student(00676676, "78575456", "Roni", "Rabideau", "09/05/2006", 2012, true),
    "11335556": new Student(11335556, "78575456", "Wendolyn", "Haber", "09/06/2006", 2012, true),
    "00827223": new Student(00827223, "78575456", "Yuonne", "Bradman", "09/07/2006", 2013, false),
    "99876655": new Student(99876655, "98071230", "Sherlene", "Melrose", "09/08/2006", 2012, true),
    "09244894": new Student(09244894, "98071230", "Felisa", "Schram", "09/09/2006", 2012, true),
    "89874378": new Student(89874378, "98071230", "Tracy", "Larose", "09/10/2006", 2013, false),
    "74723723": new Student(74723723, "98071230", "Creola", "Stanback", "09/11/2006", 2013, true),
    "09283838": new Student(09283838, "54767688", "Karri", "Dubinsky", "09/12/2006", 2012, false),
    "45675647": new Student(45675647, "54767688", "Juli", "Shelman", "09/13/2006", 2012, true),
    "92398238": new Student(92398238, "54767688", "Shelli", "Clemmons", "09/14/2006", 2013, true),
    "76656555": new Student(76656555, "54767688", "Raymonde", "Drane", "09/15/2006", 2013, true)
};

type Student object {
    float studentId;
    string schoolId;
    string firstName;
    string lastName;
    string birthDate;
    int addmissionYear;
    boolean usCitizen;

    new (studentId, schoolId, firstName, lastName, birthDate, addmissionYear, usCitizen) {
    }
};
function main(string... args) {
}
```

**Steps to reproduce:**
Run the program.
**Affected Versions:**
0.980.0

**Comment by Anoukh** [comment]
@lahirus, we have done some fixes with following PR's #9917 #9908 
Accordingly, integers and floats can not be start with `0`. Also implicit cast from int to float has been removed.
Following is the working code in current master and it will work with next release
```ballerina
import ballerina/io;
import ballerina/http;
import ballerina/log;

map<Student> studentMap = {
    "0989898": new Student(989898, "34534253", "Edwin", "Tyler", "09/01/2006", 2012, true),
    "45675647": new Student(45675647, "34534253", "Angele", "Marburger", "09/02/2006", 2012, true),
    "78687765": new Student(78687765, "34534253", "Ming", "Gower", "09/03/2006", 2013, true),
    "54667677": new Student(54667677, "34534253", "Ervin", "Rappa", "09/04/2006", 2013, false),
    "00676676": new Student(676676, "78575456", "Roni", "Rabideau", "09/05/2006", 2012, true),
    "11335556": new Student(11335556, "78575456", "Wendolyn", "Haber", "09/06/2006", 2012, true),
    "00827223": new Student(827223, "78575456", "Yuonne", "Bradman", "09/07/2006", 2013, false),
    "99876655": new Student(99876655, "98071230", "Sherlene", "Melrose", "09/08/2006", 2012, true),
    "09244894": new Student(9244894, "98071230", "Felisa", "Schram", "09/09/2006", 2012, true),
    "89874378": new Student(89874378, "98071230", "Tracy", "Larose", "09/10/2006", 2013, false),
    "74723723": new Student(74723723, "98071230", "Creola", "Stanback", "09/11/2006", 2013, true),
    "09283838": new Student(9283838, "54767688", "Karri", "Dubinsky", "09/12/2006", 2012, false),
    "45675647": new Student(45675647, "54767688", "Juli", "Shelman", "09/13/2006", 2012, true),
    "92398238": new Student(92398238, "54767688", "Shelli", "Clemmons", "09/14/2006", 2013, true),
    "76656555": new Student(76656555, "54767688", "Raymonde", "Drane", "09/15/2006", 2013, true)
};

type Student object {
    int studentId;
    string schoolId;
    string firstName;
    string lastName;
    string birthDate;
    int addmissionYear;
    boolean usCitizen;

    new (studentId, schoolId, firstName, lastName, birthDate, addmissionYear, usCitizen) {
    }
};
function main(string... args) {
}
```

**Comment by Anoukh** [comment]
Closing the issue since the fixes are in master and will be available in next release
